### ISSUE TITLE
Toggling Max-sized Object in B.prettyTuple/B.show

### ISSUE THREAD

**Issue by codykingham** [Dans-labs/text-fabric#12]
`B.prettyTuple` and `B.show` currently display results in the context of their enclosing verse. I would like to be able to request a different maximum sized object such as a sentence or clause.

Perhaps this is difficult to implement since the maximum object should fully enclose the results. If I request a maximum size of a `phrase` on a tuple containing clause nodes, what to do? One option would be to simply display all contained phrases without sentence or clause boundaries. This would look similar to the Bible Online Learner view when clause/sentence boundaries are toggled off but phrase boundaries are on. So if I toggle 'word', for instance, I get a bunch of word boxes with no phrase boundaries. 

Going up could behave similarly. If I request `clause`, then I get enclosing clauses for all attested clauses in the tuple.

Perhaps this could be toggled in `B.prettySetup`.

**Comment by codykingham** [comment]
The use case for this request is for reviewing lots of datasets. I constantly have to scroll up or down on the screen until I see my highlighted result. I would like to move through datasets faster and see the relevant result instantly.

**Comment by dirkroorda** [comment]
What I currently do is: objects outside the enclosing verse: display them as plain() would display them. Just a short identification.
Only objects of a level lower than the verse will be shown in full.

Very understandable request that the user determines the nature of the container within which pretty reigns and without which plain reigns.

I have already attempted to do this, but it turns out to be more work than expected. Two factors:

* it should work both in condensed and uncondensed mode
* the complexity of dealing with phrase and phrase atoms (idem for clauses and sentences) is biting     
  back, and what about half verses.
* then I also want to have this for cuneiform, in a way that I do not duplicate too much code.

In short, this takes a fair bit of refactoring. I'll see when I can fit it in.

**Comment by dirkroorda** [comment]
Started working on this.

Pragmatic solution for objects that are not completely contained in their container objects, such as a sentence that spans verses: I put each object in the container that contains their first slot and the container that contains their last slot. Each container cuts of the material of subobjects that stick out of the container. This will not solve cases where a sentence spans three verses, or very irregular objects.

**Comment by dirkroorda** [comment]
An other effect of condensing: objects that are not contained in any container will not show up in condensed results.

**Comment by codykingham** [comment]
What would be an example of an object like this?

**Comment by dirkroorda** [comment]
If your search for
```
verse
  sentence
    phrase function=Pred
```
and then condense to clauses, the verse and the sentence are thrown away from the condensed results.

And what should happen if you condense to `lex`, or if your search results condense `lex` items?
